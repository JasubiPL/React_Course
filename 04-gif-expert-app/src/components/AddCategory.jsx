import { useState } from 'react'

export  const AddCategory = ({onAddCategories}) => {
  
  const [inputValue, setInputValue] = useState('')

  const onInputChange = (e) =>{
    setInputValue(e.target.value);
  }

  const onSubmit = (e) =>{
    e.preventDefault();
    if(inputValue.trim().length <= 0) return;
    onAddCategories(inputValue);
    setInputValue('');
  }
  
  return(

    <form onSubmit={( onSubmit )}>
      <input 
        type='text'
        placeholder='Buscar Gifs'
        value={ inputValue }
        onChange={ onInputChange }
      />
    </form>

  )
}