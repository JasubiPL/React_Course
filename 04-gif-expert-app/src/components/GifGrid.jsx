import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) =>{
  
  const { gifData, isLoading } = useFetchGifs( category )
  // console.log({gifData, isLoading})
  
  return (
    <>
      <h3>{ category }</h3>
      <div className="card-grid">
        {isLoading ? <div className='loading'></div> :
          gifData.map(gif =>(
            <GifItem title={gif.title} image={gif.url} key={gif.id} />
          ))
        }
      </div>
      
    </>
  )
}