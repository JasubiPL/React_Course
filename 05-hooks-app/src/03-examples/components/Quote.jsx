import { useLayoutEffect, useRef, useState } from "react"

export const Quote = ({ image, name,species }) =>{
  const pRef = useRef() 
  const [boxSize, setBoxSize] = useState({ width:0, height: 0})

  useLayoutEffect(() =>{
    const { width, height } = pRef.current.getBoundingClientRect()
    setBoxSize({width, height})
  },[name])

  return(
    <>
    <blockquote 
      className='blockquote text-end'
      style={{ display: 'flex'}}
    >
      <img src={ image } alt="" srcset="" />
      <p ref={pRef} className='mb-1'>{ name}</p>
      <footer className='blockquote-footer'>{ species }</footer>
  </blockquote>
  <code>{ JSON.stringify(boxSize)}</code>
    </>
  )
}