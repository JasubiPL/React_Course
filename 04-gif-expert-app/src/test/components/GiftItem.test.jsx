import { render, screen } from '@testing-library/react'
import { GifItem } from '../../components/GifItem';

describe('Pruebas en <GiftItem.jsx />', () =>{
  
  const title = 'Jasubi'
  const url = 'https://www.google.com/saitama.jpgu'

  test('debe de hacer match con el snaptshot ', () => {
    
    
    const { container } = render(<GifItem title={ title } image={ url } />)
    expect(container).toMatchSnapshot()
  });

  test('debe de mostart la imagen con el URL y el ALT indicado', () => {
    render( <GifItem title={ title } image={ url } /> )
    // screen.debug()
    // expect( screen.getByRole('img').src ).toBe( url )
    // expect( screen.getByRole('img').alt ).toBe( title )
    const { src, alt } = screen.getByRole('img')
    expect( src ).toBe( url )
    expect( alt ).toBe( title )
  });

  test('Debe de mostrar el titulo en el componente ', () => {
    render( <GifItem title={ title } image={ url } /> )
    expect( screen.getByText( title ) ).toBeTruthy()
  });
}) 

