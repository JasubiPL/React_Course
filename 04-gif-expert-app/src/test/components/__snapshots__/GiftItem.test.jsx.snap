// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem.jsx /> debe de hacer match con el snaptshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Jasubi"
      src="https://www.google.com/saitama.jpgu"
    />
    <p>
      Jasubi
    </p>
  </div>
</div>
`;
